pandas/tests/frame/test_join.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function
+
+import numpy as np
+
+import pandas as pd
+
+from pandas.tests.frame.common import TestData
+
+import pandas.util.testing as tm
+
+
+class TestDataFrameJoin(TestData):
+
+    def test_join(self):
+        df1 = pd.DataFrame({'a': [20, 10, 0]}, index=[2, 1, 0])
+        df2 = pd.DataFrame({'b': [100, 200, 300]}, index=[1, 2, 3])
+
+        # default how='left'
+        result = df1.join(df2)
+        expected = pd.DataFrame({'a': [20, 10, 0], 'b': [200, 100, np.nan]},
+                                index=[2, 1, 0])
+        tm.assert_frame_equal(result, expected)
+
+        # how='left'
+        result = df1.join(df2, how='left')
+        expected = pd.DataFrame({'a': [20, 10, 0], 'b': [200, 100, np.nan]},
+                                index=[2, 1, 0])
+        tm.assert_frame_equal(result, expected)
+
+        # how='right'
+        result = df1.join(df2, how='right')
+        expected = pd.DataFrame({'a': [10, 20, np.nan], 'b': [100, 200, 300]},
+                                index=[1, 2, 3])
+        tm.assert_frame_equal(result, expected)
+
+        # how='inner'
+        result = df1.join(df2, how='inner')
+        expected = pd.DataFrame({'a': [20, 10], 'b': [200, 100]},
+                                index=[2, 1])
+        tm.assert_frame_equal(result, expected)
+
+        # how='outer'
+        result = df1.join(df2, how='outer')
+        expected = pd.DataFrame({'a': [0, 10, 20, np.nan],
+                                 'b': [np.nan, 100, 200, 300]},
+                                index=[0, 1, 2, 3])
+        tm.assert_frame_equal(result, expected)
+
+    def test_join_sort(self):
```

Comment:
I would like to integrate these tests and use a parametrization to make this easy; this is not part of the a class test, just a raw function.

something like

```
@pytest.fixture
def df1():
     return DataFrame(...)

@pytest.fixture
def df2():
     return DataFrame(...)

@pytest.mark.parametrize(
    "how, sort, expected", [
         ('left', False, DataFrame(.....)),
         ('left', True, DataFrame(.....))])
def test_join(how, sort, expected):

     result = df1.join(df2, how=how, sort=sort)
     tm.assert_frame_equal(result, expected)
```